### PR TITLE
FEAT: Implemented login/signup auth

### PR DESCRIPTION
**Pull request by shivanigangadharan**
- Printed signed-up user response and logged-in user on console.
- Stored the signed-up/logged-in user in context.

### CODE REVIEW

**Review by dev-enforced — APPROVED** [review]
The code for the authentication process works correctly : 
User signing up from [here](https://deploy-preview-5--shop-textiles.netlify.app/signup) can go to the [login page](https://deploy-preview-5--shop-textiles.netlify.app/login) with the same credentials to login. The log messages for these operations provide the evidence
Some minor things which you can add on while making this component :
1) Providing some errors if the user doesn't fill the details correctly while signing up like not checking the terms and conditions one
2) And a guest login account to avoid the hassle of form filling up process so that we can understand the application work
3) You can use an object while signing a user up if more fields such as name,telephone number(here only 2 are there i.e. email and password ) come into consideration in the future.

Good work @shivanigangadharan got some insights into the authentication process and how context simplifies its making

File: src/context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        })
+        try {
+
+            const userResponse = await axios.post(`/api/auth/signup`, JSON.stringify({
+                email: email,
+                password: password,
+            }));
+            if (userResponse.statusText === "Created") {
+                localStorage.setItem("token", JSON.stringify(userResponse.data.encodedToken));
+                console.log(localStorage.getItem("token"));
```
Comment (abhishektanwar) [inline src/context/authContext.js]: Please remove console logs when you add the redirection logic on successful user signup.

File: src/context/authContext.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            const userResponse = await axios.post(`/api/auth/login`, {
+                email: email,
+                password: password,
+            });
+            if (userResponse.statusText === "OK") {
+                console.log("Login successfull : ", userResponse);
+                setUser(userResponse.data.foundUser);
+            }
+            else {
+                console.log("Incorrect credentials.")
```
Comment (abhishektanwar) [inline src/context/authContext.js]: Please remove console logs when you add the redirection logic on successful user login.

**Review by abhishektanwar — APPROVED** [review]
Appreciation: The approach used in auth context is good. Clean and understandable code.

Good to merge.

**Comment by shivanigangadharan** [comment]
> The code for the authentication process works correctly : User signing up from [here](https://deploy-preview-5--shop-textiles.netlify.app/signup) can go to the [login page](https://deploy-preview-5--shop-textiles.netlify.app/login) with the same credentials to login. The log messages for these operations provide the evidence Some minor things which you can add on while making this component :
> 
> 1. Providing some errors if the user doesn't fill the details correctly while signing up like not checking the terms and conditions one
> 2. And a guest login account to avoid the hassle of form filling up process so that we can understand the application work
> 3. You can use an object while signing a user up if more fields such as name,telephone number(here only 2 are there i.e. email and password ) come into consideration in the future.
> 
> Good work @shivanigangadharan got some insights into the authentication process and how context simplifies its making

Noted Roshan, will include these suggestions in the following branches. Thanks!

**Pull request merged**